### ISSUE TITLE
Hindi Translation Progress

### ISSUE THREAD

**Issue by iliakan** [javascript-tutorial/hi.javascript.info#1]
## Maintainer List

@siradityaverma

## For New Translators

To translate a page:

1. Check that no one else has claimed your page in the checklist and comments below.
2. Comment below with the name of the page you would like to translate. **Please take only one page at a time**.
3. Clone this repo, translate your page, and submit a pull request!

Before contributing, read the glossary and style guide (once they exist) to understand how to translate various technical and React-specific terms.

Please be prompt with your translations! If you find find that you can't commit any more, let the maintainers know so they can assign the page to someone else.

## For Maintainers

When someone volunteers, edit this issue with the username of the volunteer, and with the PR. Ex:

* [ ] Home Page (@iliakan) #1

When PRs are merged, make sure to mark that page as completed!

### An introduction

* [ ] An Introduction to JavaScript
* [ ] Code editors
* [ ] Developer console

### JavaScript Fundamentals

* [ ] Hello, world!
* [ ] Code structure
* [ ] The modern mode, "use strict"
* [ ] Variables
* [ ] Data types
* [ ] Type Conversions
* [ ] Operators
* [ ] Comparisons
* [ ] Interaction: alert, prompt, confirm
* [ ] Conditional operators: if, '?'
* [ ] Logical operators
* [ ] Loops: while and for
* [ ] The "switch" statement
* [ ] Functions
* [ ] Function expressions and arrows
* [ ] JavaScript specials

### Code quality

* [ ] Debugging in Chrome
* [ ] Coding Style
* [ ] Comments
* [ ] Ninja code
* [ ] Automated testing with mocha
* [ ] Polyfills

### Objects: the basics

* [ ] Objects
* [ ] Garbage collection
* [ ] Symbol type
* [ ] Object methods, "this"
* [ ] Object to primitive conversion
* [ ] Constructor, operator "new"

### Data types

* [ ] Methods of primitives
* [ ] Numbers
* [ ] Strings
* [ ] Arrays
* [ ] Array methods
* [ ] Iterables
* [ ] Map, Set, WeakMap and WeakSet
* [ ] Object.keys, values, entries
* [ ] Destructuring assignment
* [ ] Date and time
* [ ] JSON methods, toJSON

### Advanced working with functions

* [ ] Recursion and stack
* [ ] Rest parameters and spread operator
* [ ] Closure
* [ ] The old "var"
* [ ] Global object
* [ ] Function object, NFE
* [ ] The "new Function" syntax
* [ ] Scheduling: setTimeout and setInterval
* [ ] Decorators and forwarding, call/apply
* [ ] Function binding
* [ ] Currying and partials
* [ ] Arrow functions revisited

### Object properties configuration

* [ ] Property flags and descriptors
* [ ] Property getters and setters

### Prototypes, inheritance

* [ ] Prototypal inheritance
* [ ] F.prototype
* [ ] Native prototypes
* [ ] Prototype methods, objects without __proto__
* [ ] Getting all properties

### Classes

* [ ] Class basic syntax
* [ ] Class inheritance
* [ ] Static properties and methods
* [ ] Private and protected properties and methods
* [ ] Extending built-in classes
* [ ] Class checking: "instanceof"
* [ ] Mixins

### Error handling

* [ ] Error handling, "try..catch"
* [ ] Custom errors, extending Error

### Promises, async/await

* [ ] Introduction: callbacks
* [ ] Promise
* [ ] Promises chaining
* [ ] Error handling with promises
* [ ] Promise API
* [ ] Promisification
* [ ] Microtasks and event loop
* [ ] Async/await

### Generators, advanced iteration

* [ ] Generators
* [ ] Async iterators and generators

### Modules

* [ ] Modules, introduction
* [ ] Export and Import
* [ ] Dynamic imports

### Document

* [ ] Browser environment, specs
* [ ] DOM tree
* [ ] Walking the DOM
* [ ] Searching: getElement*, querySelector*
* [ ] Node properties: type, tag and contents
* [ ] Attributes and properties
* [ ] Modifying the document
* [ ] Styles and classes
* [ ] Element size and scrolling
* [ ] Window sizes and scrolling
* [ ] Coordinates

### Introduction to Events

* [ ] Introduction to browser events
* [ ] Bubbling and capturing
* [ ] Event delegation
* [ ] Browser default actions
* [ ] Dispatching custom events

### UI Events

* [ ] Mouse events basics
* [ ] Moving: mouseover/out, mouseenter/leave
* [ ] Drag'n'Drop with mouse events
* [ ] Keyboard: keydown and keyup
* [ ] Scrolling

### Forms, controls

* [ ] Form properties and methods
* [ ] Focusing: focus/blur
* [ ] Events: change, input, cut, copy, paste
* [ ] Form submission: event and method submit

### Document and resource loading

* [ ] Page: DOMContentLoaded, load, beforeunload, unload
* [ ] Scripts: async, defer
* [ ] Resource loading: onload and onerror

## Frames and windows

* [ ] Popups and window methods
* [ ] Cross-window communication
* [ ] The clickjacking attack

## Binary data, files

* [ ] ArrayBuffer, binary arrays
* [ ] TextDecoder and TextEncoder
* [ ] Blob
* [ ] File and FileReader

## Network requests

* [ ] Fetch: Basics
* [ ] Fetch: Download progress
* [ ] Fetch: Abort
* [ ] Fetch: Cross-Origin Requests
* [ ] Fetch API
* [ ] URL objects
* [ ] XMLHttpRequest
* [ ] WebSocket

## Storing data in the browser

* [ ] Cookies, document.cookie
* [ ] LocalStorage, sessionStorage
* [ ] IndexedDB

## Animation

* [ ] Bezier curve
* [ ] CSS-animations
* [ ] JavaScript animations

## Web components

* [ ] From the orbital height
* [ ] Custom elements
* [ ] Shadow DOM
* [ ] Template element
* [ ] Shadow DOM slots, composition
* [ ] Shadow DOM styling
* [ ] Shadow DOM and events

## Regular expressions

* [ ] Patterns and flags
* [ ] Methods of RegExp and String
* [ ] Character classes
* [ ] Escaping, special characters
* [ ] Sets and ranges [...]
* [ ] Quantifiers +, *, ? and {n}
* [ ] Greedy and lazy quantifiers
* [ ] Capturing groups
* [ ] Backreferences in pattern: \n and \k
* [ ] Alternation (OR) |
* [ ] String start ^ and finish $
* [ ] Multiline mode, flag "m"
* [ ] Lookahead and lookbehind
* [ ] Infinite backtracking problem
* [ ] Unicode: flag "u"
* [ ] Unicode character properies \p
* [ ] Sticky flag "y", searching at position

## Miscellaneous

* [ ] Mutation observer

**Comment by sutharanand17** [comment]
Debugging in Chrome

**Comment by mittalyashu** [comment]
Seems like first one `An Introduction to JavaScript` is already localized in Hindi language but it's still unchecked.

**Question:** Is this issue is about translation or localization because both of them are very different things?
> I am a member of Kubernetes team for localising content in hindi language.

**Comment by iliakan** [comment]
That's because PR didn't have the same title, so the bot didn't check it.
And also the maintainer didn't check it here.

**Comment by iliakan** [comment]
@siradityaverma are you using the bot?

**Comment by prarora98** [comment]
I want to translate JavaScript Fundamentals whole section

**Comment by iliakan** [comment]
Please take 1 article at a time.

**Comment by prarora98** [comment]
I can do Hello, world! @iliakan

**Comment by siradityaverma** [comment]
@iliakan sorry for the late reply. My GitHub notifications are flooded so I missed the PR notifications from the bot. Is there a dedicated slack channel or something for this?

**Comment by iliakan** [comment]
Personally I made some email filters for my client :-)

**Comment by verv0eren** [comment]
An Introduction to JavaScript

**Comment by mayank-gangwal** [comment]
I'd like to take translation of **Manuals and specifications**

**Comment by Harendra1993** [comment]
Code editors @iliakan

**Comment by iliakan** [comment]
If @siradityaverma isn't online any more, I could assign another collaborator, to be able to review and merge PRs.

Let me know if your JS knowledge is good enough for that.

**Comment by Harendra1993** [comment]
@iliakan  I am happy to do that.

**Comment by siradityaverma** [comment]
@Harendra1993 / @iliakan if you want to take over this project, that'd be great. I've been so busy that I am not able to contribute anything apart from reviewing PRs and merge requests. Just be careful with the PRs. People raise requests by directly translating from translation tools and the grammar just ends up being completely wrong. :)

**Comment by iliakan** [comment]
Done!

**Comment by theBstar** [comment]
I don't see any progress being made here. Can I start on this?

**Comment by iliakan** [comment]
@siradityaverma @Harendra1993 really did nothing?

**Comment by theBstar** [comment]
> @siradityaverma @Harendra1993 really did nothing?

Almost yes. I can't see any changes on the checklist as well as on the repo. There is only a small chunk of a file changed which also could have been done well. As @siradityaverma mentioned seems he is not having enough time. I can take this from here.

**Comment by iliakan** [comment]
@theBstar you're the maintainer now.

**Comment by theBstar** [comment]
> I'd like to take translation of **Manuals and specifications**

@mayank-gangwal Sorry for the late reply. Please confirm if you want to pick this up. I will assign it to you.

**Comment by ayushhagarwal** [comment]
I wish to translate code editors.

**Comment by theBstar** [comment]
> I wish to translate code editors.

Thanks for your interest @ayushhagarwal . I am assigning you the article then. Let me know if you need anything. Also, we are preparing a glossary to follow, It will be updated on other PR. I will tag you there as well.

**Comment by ayushhagarwal** [comment]
hey @theBstar after forking should i make changes in the master branch itself and submit the pull request from the same master branch?

**Comment by theBstar** [comment]
> hey @theBstar after forking should i make changes in the master branch itself and submit the pull request from the same master branch?

Works @ayushhagarwal

**Comment by theBstar** [comment]
An Introduction to JavaScript

**Comment by ayushhagarwal** [comment]
Hey @theBstar I would like to take up manuals and specifications next.

**Comment by theBstar** [comment]
> Manuals and specifications

Great. If you comment the exact title, the bot will assign the article to you if its not already taken. I have updated the comment and assigned the article to you.

**Comment by theBstar** [comment]
Hello, world!

**Comment by theBstar** [comment]
Developer console

**Comment by theBstar** [comment]
Code structure

**Comment by ashpratap007** [comment]
Variables

**Comment by amit11893** [comment]
Prototypal inheritance

**Comment by theBstar** [comment]
Thanks @amit11893 . Let us know if you face any difficulties or need help with anything.

**Comment by ashpratap007** [comment]
Data types

**Comment by anil-maurya** [comment]
Interaction: alert, prompt, confirm

**Comment by ashpratap007** [comment]
> Error: the article already has meta information in the Progress Issue "(@ashpratap007)" ⁉️

I got the following error on Choosing "Data Type". Can someone guide me, if I can proceed or not?

**Comment by theBstar** [comment]
> > Error: the article already has meta information in the Progress Issue "(@ashpratap007)" interrobang
> 
> I got the following error on Choosing "Data Type". Can someone guide me, if I can proceed or not?

Hi, Don't worry about the error. Please proceed with the MR.

**Comment by rjitsu** [comment]
Comparisions

**Comment by ayushhagarwal** [comment]
Ninja Code

**Comment by tiwarirr** [comment]
Basic operators, maths

**Comment by PraveenPal4232** [comment]
Hello team I would like to translate the "Type Conversions" Article

**Comment by Athira2199** [comment]
Loops: while and for

**Comment by Athira2199** [comment]
The "switch" statement

**Comment by theBstar** [comment]
> Hello team I would like to translate the "Type Conversions" Article

@PraveenPal4232 I am assigning the article to you. You can just copy the title from the checklist and comment, It will be auto assigned. Thanks.

**Comment by AabhasSao** [comment]
**Conditional branching: if, '?'** 
_I would like to translate this._
I like this resource very much and it would be so great to have this resource in hindi.
there isn't much activity here although. Maybe we can make progress

**Comment by theBstar** [comment]
> **Conditional branching: if, '?'**
> _I would like to translate this._
> I like this resource very much and it would be so great to have this resource in hindi.
> there isn't much activity here although. Maybe we can make progress

I am assigning the issue to you. Here onwards, You can just copy the title from checklist above and the bot will auto assign the article to you.

**Comment by theBstar** [comment]
> **Conditional branching: if, '?'**
> _I would like to translate this._
> I like this resource very much and it would be so great to have this resource in hindi.
> there isn't much activity here although. Maybe we can make progress

I haven't been able to dedicate enough time nowadays. If anyone is interested we can add them as maintainer. I can still help out with translation and reviews.

**Comment by nik2490** [comment]
Conditional branching: if, '?'

**Comment by Nirbhayparmar** [comment]
Logical operators

**Comment by Nirbhayparmar** [comment]
Can any guide me for this translation because this is my first time contribution here?

**Comment by Nirbhayparmar** [comment]
@theBstar can you help me? I don't know how to begin the process of translation?

**Comment by theBstar** [comment]
> Can any guide me for this translation because this is my first time contribution here?

1. You can fork this Repo
2. Update the required file -> the chapter, question, and answer (only the non-code part)
3. Make a pull request

To update the required file, I usually use the google translator tool, then go through the machine translation and make any changes required.

If this is your first ever pull request, https://www.freecodecamp.org/news/how-to-make-your-first-pull-request-on-github-3/ Link above might help. (the article above shows git usage via command line, but for this project you can mostly rely on the github gui)

**Comment by Nirbhayparmar** [comment]
No, this not my first pull request. But I am first time contributing to such a large code base so wanted start carefully, I don't want to mess up.

**Comment by Nirbhayparmar** [comment]
So

**Comment by incrediblenura** [comment]
_**Comments**_

**Comment by shockz09** [comment]
Code structure

**Comment by shockz09** [comment]
The modern mode, "use strict"

**Comment by shockz09** [comment]
Variables

**Comment by shockz09** [comment]
Functions

**Comment by shockz09** [comment]
Code editors

**Comment by rskhan167** [comment]
Functions

**Comment by rskhan167** [comment]
Function expressions

**Comment by rskhan167** [comment]
> > **Conditional branching: if, '?'**
> > _I would like to translate this._
> > I like this resource very much and it would be so great to have this resource in hindi.
> > there isn't much activity here although. Maybe we can make progress
> 
> I haven't been able to dedicate enough time these days. If anyone is interested we can add them as maintainer. I can still help out with translation and reviews.

Hi @theBstar, I am interested in maintaining this project. Please let me know if this is possible? Thanks.

**Comment by iliakan** [comment]
@rskhan167 Is your JS knowledge good enough to review PRs?

**Comment by iliakan** [comment]
Maintainers are welcome, Hindi is language of many people.

**Comment by rskhan167** [comment]
Yes, @iliakan I think so. I am working as a Nodejs developer for the past year.

**Comment by iliakan** [comment]
Done! You're the maintainer now, @rskhan167 !

**Comment by rskhan167** [comment]
Thanks @iliakan

**Comment by Deep1Shikha** [comment]
Comments

**Comment by Deep1Shikha** [comment]
Polyfills and transpilers

**Comment by Deep1Shikha** [comment]
@iliakan , We should encourage people to contribute in this repository by participating in Hacktoberfest. It will boost engagement.

**Comment by theBstar** [comment]
Yes agree to some extent. But concerned about the quality of work. There
are still 2 hanging PR from last October fest. And most of these people are
students or starting out. We need people who actually care about the js
community and have decent knowledge of js to do the translation work. Just
my personal experience and personal views.

**Comment by iliakan** [comment]
@Deep1Shikha We tried, it doesn't work. Commits from Hacktoberfest are crap.

**Comment by Deep1Shikha** [comment]
@iliakan @theBstar Agreed. Quality is utmost important for us.

**Comment by rskhan167** [comment]
Arrow functions, the basics

**Comment by iamcreed** [comment]
i would like to translate " Objects" @iliakan

**Comment by oreo3494** [comment]
I'm willing to translate "[Data Types](https://github.com/javascript-tutorial/hi.javascript.info/blob/master/1-js/02-first-steps/05-types/article.md)" article

**Comment by oreo3494** [comment]
Data types

**Comment by oreo3494** [comment]
Nullish coalescing operator '??'

**Comment by thenil3sh** [comment]
Nullish coalescing operator '??'

**Comment by AKR-2803** [comment]
Objects @iliakan  @a4averma

**Comment by divyaprakashdp** [comment]
[Object references and copying](https://github.com/javascript-tutorial/hi.javascript.info/blob/master/1-js/04-object-basics/02-object-copy)

**Comment by bishuk-dev** [comment]
Methods of primitive

**Comment by swapnasahoo** [comment]
I want to translate [JavaScript specials](https://github.com/javascript-tutorial/hi.javascript.info/blob/master/1-js/02-first-steps/18-javascript-specials)

**Comment by iliakan** [comment]
@rskhan167 are you available to accept the PRs?

**Comment by rskhan167** [comment]
> [@rskhan167](https://github.com/rskhan167) are you available to accept the PRs?

yes @iliakan I am available

**Comment by iliakan** [comment]
@rskhan167 👍 

@swapnasahoo wrote me requesting to become a contributor, did you see his PRs?

**Comment by nanhaayurveda-kpt** [comment]
Object methods, "this"
